### DISCUSSION TITLE
Single-shard cluster

### DISCUSSION THREAD

**Discussion by zygisa** [valkey-io/valkey#870] [General]
👋 Hello valkey community,

I have a few questions about single-shard cluster and Sentinel.

- Are you planning to add single-shard cluster functionality in the next major release? If not, when can we expect it?
- Is Sentinel going to be deprecated once single-shard cluster feature is out?

Thanks!

**Comment by madolson** [comment]
> Are you planning to add single-shard cluster functionality in the next major release? If not, when can we expect it?
Is Sentinel going to be deprecated once single-shard cluster feature is out?

The feature will definitely not be available in Valkey 8.0, maybe a future version though.

**Reply by hpatro** [reply]
@madolson Is it dependent on V2 implementation or can it be achieved separately? What's the feature gap in supporting this  (voting replicas for failover ?)

**Reply by madolson** [reply]
We could achieve it separately. We just need some way to indicate that a replica is able to vote as part of the quorum group.

**Comment by zuiderkwast** [comment]
Single-shard cluster is already possible, but it's not able to do automatic failovers. If you have some other control plane that triggers a manual failover, it can work, but I guess it's not what you want. :)

To be able to perform automatic failovers, the cluster needs at least 3 voting nodes, so we can allow replicas to vote, at least under certain conditions. It's important that all nodes in the cluster agree about who can vote and who can't, so they need to negotiate this information in some way (using some unused bits in the cluster bus protocol).
